packages/pwa/app/commerce-api/auth.js

```diff
@@ -236,18 +229,18 @@ class Auth {
             throw new HTTPError(response.status, json.message)
         }
 
-        const tokenBody = createGetTokenBody(
-            response.url,
-            `${getAppOrigin()}${slasCallbackEndpoint}`,
-            window.sessionStorage.getItem('codeVerifier')
-        )
+        let tokenURL = response.url
+        if (!this._onClient) {
+            tokenURL = response.headers.get('location')
```

Comment:
Perhaps add a comment here to inform that elsewhere we've configured fetch to not follow the redirect. So in this case, `response.url` would be the url of the original request. To get the _final_ url, we can get it from the location header instead.